lib/Driver/DarwinToolChains.cpp

```diff
@@ -392,6 +393,35 @@ toolchains::Darwin::constructInvocation(const LinkJobAction &job,
   SmallString<128> RuntimeLibPath;
   getRuntimeLibraryPath(RuntimeLibPath, context.Args, /*Shared=*/true);
 
+  // Link the compatibility library, if we're deploying back to OSes that
+  // have a Swift 5.0 runtime.
+  
+  bool backDeployToSwift50;
+  unsigned Major, Minor, Micro;
+  
+  if (Triple.isMacOSX()) {
+    Triple.getMacOSXVersion(Major, Minor, Micro);
+    backDeployToSwift50 = Minor <= 14;
+  } else if (Triple.isiOS()) {
+    Triple.getiOSVersion(Major, Minor, Micro);
+    backDeployToSwift50 = Major <= 12;
+  } else if (Triple.isWatchOS()) {
+    Triple.getWatchOSVersion(Major, Minor, Micro);
+    backDeployToSwift50 = Major <= 5;
+  } else {
+    backDeployToSwift50 = false;
+  }
+  if (backDeployToSwift50) {
+    SmallString<128> BackDeployLib;
+    BackDeployLib.append(RuntimeLibPath);
+    llvm::sys::path::append(BackDeployLib, "libswiftCompatibility50.a");
+
+    if (llvm::sys::fs::exists(BackDeployLib)) {
+      Arguments.push_back("-force_load");
+      Arguments.push_back(context.Args.MakeArgString(BackDeployLib));
```

Comment:
Should this only happen in executable targets, or everywhere? What happens when there's a zillion copies of this section in the final app?

Reply:
It only takes effect in the executable. I should change it so we only link it (by default) into executables.